### PR TITLE
Update download path

### PR DESCRIPTION
**Pull request by mairaw**
I've changed the download path on the marketing site today from dotnet-core to dotnet. Everything should be redirecting but I like to always use the proper target URL.

And I've made some small changes where I saw some issues where the changes happened.

### CODE REVIEW

**Review by sywhang — APPROVED** [review]
changes to the diagnostics docs LGTM

File: samples/snippets/standard/async/async-and-await/vb/readme.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ The code for the *MainWindow.xaml.vb* file from this sample is included in the a
 
 ## Sample prerequisites
 
-This sample is written in Visual Basic and targets .NET Core 3.0 running on Windows. It requires the [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.0).
+This sample is written in Visual Basic and targets .NET Core 3.0 running on Windows. It requires the [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet/3.0).
```
Comment (Youssef1313) [inline samples/snippets/standard/async/async-and-await/vb/readme.md:24]: While you're at it, the sample is now using 3.1

https://github.com/dotnet/docs/blob/327f2c839b2cd78d3b22057963304ec06d9713c8/samples/snippets/standard/async/async-and-await/vb/AsyncFirstExampleVB.vbproj#L5

File: samples/snippets/standard/async/async-and-await/cs/readme.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ The code for the *MainWindow.xaml.cs* file from this sample is included in the a
 
 ## Sample prerequisites
 
-This sample is written in C# and targets .NET Core 3.0 running on Windows. It requires the [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.0).
+This sample is written in C# and targets .NET Core 3.0 running on Windows. It requires the [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet/3.0).
```
Comment (Youssef1313) [inline samples/snippets/standard/async/async-and-await/cs/readme.md:24]: Now using 3.1:

https://github.com/dotnet/docs/blob/327f2c839b2cd78d3b22057963304ec06d9713c8/samples/snippets/standard/async/async-and-await/cs/AsyncFirstExampleCS.csproj#L5

**Comment by mairaw** [comment]
Good catch. Thanks @Youssef1313.

**Review by BillWagner — APPROVED** [review]
This LGTM @mairaw 

I'll :shipit: now.

I apologize for the delay.

**Pull request merged**